requirements/dev/dev-requirements.txt

```diff
@@ -1,3 +1,4 @@
+-r ../build/build-requirements.txt
```

Comment:
@tacaswell I think this is where the CI is failing as it is trying to install pybind11 in editable mode. But CI doesn't install those in editable mode. 

Does it make sense to separate this out of the -e flag?